Source/WebCore/rendering/RenderThemeCocoa.mm

```diff
@@ -243,8 +243,33 @@ static inline FontSelectionValue cssWeightOfSystemFont(CTFontRef font)
 
 #endif
 
-static Color grammarColor(bool useDarkMode)
+Color RenderThemeCocoa::platformSpellingMarkerColor(OptionSet<StyleColorOptions> options) const
 {
+    auto useDarkMode = options.contains(StyleColorOptions::UseDarkAppearance);
+    return useDarkMode ? SRGBA<uint8_t> { 255, 140, 140, 217 } : SRGBA<uint8_t> { 255, 59, 48, 191 };
+}
+
+Color RenderThemeCocoa::platformDictationAlternativesMarkerColor(OptionSet<StyleColorOptions> options) const
+{
+    auto useDarkMode = options.contains(StyleColorOptions::UseDarkAppearance);
+    return useDarkMode ? SRGBA<uint8_t> { 40, 145, 255, 217 } : SRGBA<uint8_t> { 0, 122, 255, 191 };
```

Comment:
Is there a way we can factor out all the same special numbers so that it's more clear that they're all the same thing? I'm also assuming there's no way to get these from the system or elsewhere and they need to be manually defined.

Reply:
I was doing some digging and it appears that there are system colors that we can use for these!

Though I will need to audit them individually to compare behavior. Since this hardcoding is not new I have filed https://bugs.webkit.org/show_bug.cgi?id=252868 to address this separately.

For now, I have updated the PR to unify the dictation alternative paths, and share the same color method, reducing the amount of duplication.